### DISCUSSION TITLE
Does JuliaPOMDP support solving MDPs without discounting?

### DISCUSSION THREAD

**Discussion by yaronshap** [JuliaPOMDP/POMDPs.jl#515] [Q&A]
Hello, 

Does JuliaPOMDP supports solving MDPs without discounting? that is, optimizing long-run average costs? (I couldn't find such examples on the website).

Thank you,
Yaron

**Comment by zsunberg** [comment]
POMDPs.jl assumes that the objective is the discounted sum of future rewards and there are currently no solvers that will optimize the average reward objective.

If you are interested in writing a solver, that would be a great contribution!

(For other people reading this, you can set the discount factor to 1, but this will only work if there are absorbing terminal states, for example if there is a finite horizon (https://github.com/JuliaPOMDP/FiniteHorizonPOMDPs.jl). But this is not what the original post was asking.)

**Comment by yaronshap** [comment]
Thank you so much for the quick response, I appreciate it. 

Definitely, I'll be happy to share any code that I write. Not sure though that my coding practices meet your standards. Will such code be reviewed and is there someone who could potentially advise about improving it? 

Thanks again,
Yaron

**Reply by yaronshap** [reply]
By the way, amazing package!

**Reply by mykelk** [reply]
Great! Yep, PRs will be carefully reviewed. Thanks!
